src/cli/rustup_mode.rs

```diff
@@ -793,44 +795,67 @@ async fn default_(
 
 async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<utils::ExitCode> {
     let mut update_available = false;
-
-    let mut t = cfg.process.stdout().terminal(cfg.process);
     let channels = cfg.list_channels()?;
-
-    for channel in channels {
-        let (name, distributable) = channel;
-        let current_version = distributable.show_version()?;
-        let dist_version = distributable.show_dist_version().await?;
-        let _ = t.attr(terminalsource::Attr::Bold);
-        write!(t.lock(), "{name} - ")?;
-        match (current_version, dist_version) {
-            (None, None) => {
-                let _ = t.fg(terminalsource::Color::Red);
-                writeln!(t.lock(), "Cannot identify installed or update versions")?;
-            }
-            (Some(cv), None) => {
-                let _ = t.fg(terminalsource::Color::Green);
-                write!(t.lock(), "Up to date")?;
-                let _ = t.reset();
-                writeln!(t.lock(), " : {cv}")?;
-            }
-            (Some(cv), Some(dv)) => {
-                update_available = true;
-                let _ = t.fg(terminalsource::Color::Yellow);
-                write!(t.lock(), "Update available")?;
-                let _ = t.reset();
-                writeln!(t.lock(), " : {cv} -> {dv}")?;
-            }
-            (None, Some(dv)) => {
-                update_available = true;
-                let _ = t.fg(terminalsource::Color::Yellow);
-                write!(t.lock(), "Update available")?;
-                let _ = t.reset();
-                writeln!(t.lock(), " : (Unknown version) -> {dv}")?;
-            }
-        }
+    let num_channels = channels.len();
+    if num_channels > 0 {
+        let mp = MultiProgress::with_draw_target(ProgressDrawTarget::stdout());
+
+        let progress_bars: Vec<_> = channels
+            .iter()
+            .map(|(name, _)| {
+                let pb = mp.add(ProgressBar::new(1));
+                pb.set_style(
+                    ProgressStyle::with_template("{msg} {spinner:.green}")
+                        .unwrap()
+                        .tick_chars("⠁⠂⠄⡀⢀⠠⠐⠈ "),
+                );
+                pb.set_message(format!("{} - Checking...", name));
+                pb.enable_steady_tick(std::time::Duration::from_millis(100));
+                pb
+            })
+            .collect();
+
+        let channels = tokio_stream::iter(channels.into_iter().zip(progress_bars))
```

Comment:
Why are `progress_bars` and `channels` created in separate loops? Suggest just using for-loops here instead of a trivial combinator chain with a long `map()` closure.